src/main/java/teammates/common/util/AdminLogQuery.java

```diff
@@ -0,0 +1,175 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.exception.InvalidParametersException;
+
+import com.google.appengine.api.log.LogQuery;
+import com.google.appengine.api.log.LogService.LogLevel;
+
+/**
+ * A wrapper class for LogQuery to retrieve logs from GAE server.
+ */
+public class AdminLogQuery {
+    /**
+     * Maximum number of versions to query.
+     * The current value will include the current version and its 5 preceding versions.
+     */
+    private static final int MAX_VERSIONS_TO_QUERY = 6;
+    
+    /**
+     * A flag to decide whether to include application logs in result or not.
+     */
+    private static final boolean INCLUDE_APP_LOG = true;
+    
+    /**
+     * Affects the internal strategy to get logs. It doesn't affect the result.
+     */
+    private static final int BATCH_SIZE = 1000;
+    private static final LogLevel MIN_LOG_LEVEL = LogLevel.INFO;
+    
+    private LogQuery query;
+    private Long endTime;
+    private List<String> versionList;
+    
+    /**
+     * Default constructor.
+     */
+    public AdminLogQuery() {
+        query = LogQuery.Builder.withDefaults();
+        query.includeAppLogs(INCLUDE_APP_LOG);
+        query.batchSize(BATCH_SIZE);
+        query.minLogLevel(MIN_LOG_LEVEL);
+    }
+    
+    /**
+     * Sets values for query.
+     * If versionsToQuery is null or empty, default versions will be used instead.
+     * 
+     * @param versionsToQuery 
+     * @param startTime
+     * @param endTime
+     */
+    public AdminLogQuery(List<String> versionsToQuery, Long startTime, Long endTime) {
+        this();
+        setTimePeriodForQuery(startTime, endTime);
+        versionList = getVersionIdsForQuery(versionsToQuery);
+        query.majorVersionIds(versionList);
+    }
+    
+    /**
+     * Gets query to retrieve logs.
+     */
+    public LogQuery getQuery() {
+        return query;
+    }
+    
+    /**
+     * Sets time period to search for query.
+     * @param startTime
+     * @param endTime
+     */
+    private void setTimePeriodForQuery(Long startTime, Long endTime) {
+        if (startTime != null) {
+            query.startTimeMillis(startTime);
+        }
+        if (endTime != null) {
+            query.endTimeMillis(endTime);
+            setEndTime(endTime);
+        }
+    }
+    
+    /**
+     * Sets time period to query logs then moves endTime back by the same amount.
+     * @param timeInMillis milliseconds before endTime
+     */
+    public void setQueryWindowBackward(long timeInMillis) {
+        if (getEndTime() == null) {
+            setEndTime(TimeHelper.now(0.0).getTimeInMillis());
+        }
```

Comment:
Is there any need to keep the end time as null? If there isn't, it can be set in the constructor.